commands/check_cloudformation.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python
+import os
+from pathlib import Path
+from subprocess import run
+
+import click
+
+from commands.bootstrap_cli import make_config
+from commands.exceptions.CheckCloudformationFailure import CheckCloudformationFailure
+from commands.copilot_cli import make_storage
+
+BASE_DIR = Path(__file__).parent.parent
+
+
+def valid_checks():
+    return {
+        "lint": lint,
+    }
+
+
+@click.group(invoke_without_command=True)
+@click.argument("checks", nargs=-1)
+@click.pass_context
+def check_cloudformation(ctx, checks):
+    """Runs the checks passed in the command arguments.\n
+    If no argument is passed, it will run all the checks.
+    """
+
+    if len(checks) == 0:
+        checks = valid_checks()
+        running_checks = "all"
+    else:
+        running_checks = ' & '.join(', '.join(checks).rsplit(', ', 1))
+
+    for check in checks:
+        if check not in valid_checks():
+            click.secho(f"""Invalid check requested "{check}" """, fg="red")
```

Comment:
You could leverage click to do this validation and render the help page etc.

Reply:
Outputting the help is a good call.

I don't think we can specify what the valid arguments are for clieck when we're allowing any number from 0 to many...

https://click.palletsprojects.com/en/8.1.x/arguments/

Reply:
In the demo I mocked up, it already does this, including pulling in the sub command doc strings and rendering those on the help page

```
./clicktest.py --help
Usage: clicktest.py [OPTIONS] COMMAND1 [ARGS]... [COMMAND2 [ARGS]...]...

Options:
  --help  Show this message and exit.

Commands:
  cfn   This command generates cfn
  lint  This command lints code
  waf   this verifies the WAF exists
```

And if you enter the wrong command:

```
./clicktest.py lintxx
Usage: clicktest.py [OPTIONS] COMMAND1 [ARGS]... [COMMAND2 [ARGS]...]...
Try 'clicktest.py --help' for help.

Error: No such command 'lintxx'.
```